public/app/plugins/panel/geomap/layers/data/routeLayer.tsx

```diff
@@ -147,9 +145,38 @@ export const routeLayer: MapLayerRegistryItem<RouteConfig> = {
                   flowStyle.setArrowSize((arrowSize1 ?? 0) * 1.5);
                 }
               }
-              const LS = new LineString([coordinates[i], coordinates[i + 1]]);
-              flowStyle.setGeometry(LS);
-              styles.push(flowStyle);
+              const LS = new LineString([coordinates[j], coordinates[i + 1]]);
+              //TODO: let's try to find a less intensive check
+              const pixel1 = map.getPixelFromCoordinate(coordinates[j]);
+              const pixel2 = map.getPixelFromCoordinate(coordinates[i + 1]);
+              const deltaX = Math.abs(pixel1[0] - pixel2[0]);
+              const deltaY = Math.abs(pixel1[1] - pixel2[1]);
+              // Only render segment if change in pixel coordinates is significant enough
+              if (deltaX > 2 || deltaY > 2) {
+                flowStyle.setGeometry(LS);
+                styles.push(flowStyle);
+                j = i + 1;
```

Comment:
This logic is a little hard for me to follow / understand, what index is `j` tracking? Are we basically determining at what resolution it is necessary to draw a line vs a single point or?

This logic may be clearer if we broke it out into a util func with a few unit tests for good measure 😃

Reply:
Great suggestion, will pull it out and try to simplify!